utils/src/main/java/ru/nms/labs/TestDataGenerator.java

```diff
@@ -0,0 +1,18 @@
+package ru.nms.labs;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class TestDataGenerator {
+    public static List<Integer> generateRandomList(int size, int randSeed) {
+            List<Integer> randomList = new ArrayList<>();
```

Comment:
```suggestion
            List<Integer> randomList = new ArrayList<>(size);
```

Тогда можно избежать реаллокаций